src/Swarm/TUI/Model.hs

```diff
@@ -20,20 +20,36 @@ module Swarm.TUI.Model (
   Modal (..),
 
   -- * UI state
+
+  -- ** REPL
   REPLHistItem (..),
-  firstReplEntry,
+  replItemText,
+  isREPLEntry,
+  getREPLEntry,
+  REPLHistory,
+  replIndex,
+  newREPLHistory,
+  addREPLEntry,
+  addREPLOutput,
+  replLength,
```

Comment:
I think that if we agree on the proposed refactoring for `adjReplHistIndex` there would be no need to export `replLength` anymore.